File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int jumps = 0, farthest = 0, end = 0;
        for (int i = 0; i < n - 1; ++i) {
            farthest = max(farthest, i + nums[i]);
            if (farthest >= n - 1) {
                ++jumps;
                break;
            }
            if (i == end) {
                ++jumps;
                end = farthest;
            }
        }
        return jumps;
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int j=0;
        int currEnd=0;
        int far=0;
        for(int i=0;i<nums.length-1;i++){
            far=Math.max(far,i+nums[i]);
            if(i==currEnd){
                j+=1;
                currEnd=far;
            }
        }
        return j;
    }
}